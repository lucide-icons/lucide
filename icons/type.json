{
  "$schema": "../icon.schema.json",
  "contributors": [
    "colebemis",
    "ericfennis"
  ],
  "tags": [
    "text",
    "font",
    "typography"
  ],
  "categories": [
    "text"
  ]
}