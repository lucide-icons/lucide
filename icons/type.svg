<svg
  xmlns="http://www.w3.org/2000/svg"
  width="24"
  height="24"
  viewBox="0 0 24 24"
  fill="none"
  stroke="currentColor"
  stroke-width="2"
  stroke-linecap="round"
  stroke-linejoin="round"
>
  <path d="M12 4v16" />
  <path d="M4 7V5a1 1 0 0 1 1-1h14a1 1 0 0 1 1 1v2" />
  <path d="M9 20h6" />
</svg>
