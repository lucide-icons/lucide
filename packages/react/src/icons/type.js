import React, { forwardRef } from "react";
import PropTypes from "prop-types";

const Type = forwardRef(
  ({ color = "currentColor", size = 24, ...rest }, ref) => {
    return (
      <svg
        ref={ref}
        xmlns="http://www.w3.org/2000/svg"
        width={size}
        height={size}
        viewBox="0 0 24 24"
        fill="none"
        stroke={color}
        strokeWidth="2"
        strokeLinecap="round"
        strokeLinejoin="round"
        {...rest}
      >
        <polyline points="4 7 4 4 20 4 20 7"></polyline>
        <line x1="9" y1="20" x2="15" y2="20"></line>
        <line x1="12" y1="4" x2="12" y2="20"></line>
      </svg>
    );
  }
);

Type.propTypes = {
  color: PropTypes.string,
  size: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
};

Type.displayName = "Type";

export default Type;
