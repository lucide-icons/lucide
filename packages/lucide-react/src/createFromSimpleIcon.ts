import createLucideIcon from './createLucideIcon';
import type { IconNode } from './types';
import type { SimpleIcon } from 'simple-icons';

const createIconNodeFromSimpleIcon = (simpleIcon: SimpleIcon): IconNode => [
  [
    'path',
    {
      d: simpleIcon.path,
      fill: 'currentColor',
    },
  ],
];

/**
 * Converts a simple-icon into a Lucide icon component
 * @param name The name of the icon (will be used as component name)
 * @param simpleIcon The simple-icon object
 * @returns A Lucide icon component
 */
const createFromSimpleIcon = (name: string, simpleIcon: SimpleIcon) =>
  createLucideIcon(name, createIconNodeFromSimpleIcon(simpleIcon));

export default createFromSimpleIcon;
