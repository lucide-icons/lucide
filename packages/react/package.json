{
  "name": "lucide-react",
  "version": "1.0.0",
  "description": "React component for lucide icons",
  "main": "src/index.js",
  "typings": "src/index.d.ts",
  "author": "John Letey",
  "license": "ISC",
  "scripts": {
    "compile": "rm -rf src/icons && node bin/build.js"
  },
  "devDependencies": {
    "cheerio": "^1.0.0-rc.3",
    "prettier": "^2.0.5",
    "uppercamelcase": "^3.0.0"
  }
}
