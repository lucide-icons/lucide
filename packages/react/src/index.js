export { default as Activity } from "./icons/activity";
export { default as Airplay } from "./icons/airplay";
export { default as AlertCircle } from "./icons/alert-circle";
export { default as AlertOctagon } from "./icons/alert-octagon";
export { default as AlertTriangle } from "./icons/alert-triangle";
export { default as AlignCenter } from "./icons/align-center";
export { default as AlignJustify } from "./icons/align-justify";
export { default as AlignLeft } from "./icons/align-left";
export { default as AlignRight } from "./icons/align-right";
export { default as Anchor } from "./icons/anchor";
export { default as Aperture } from "./icons/aperture";
export { default as Archive } from "./icons/archive";
export { default as ArrowDownCircle } from "./icons/arrow-down-circle";
export { default as ArrowDownLeft } from "./icons/arrow-down-left";
export { default as ArrowDownRight } from "./icons/arrow-down-right";
export { default as ArrowDown } from "./icons/arrow-down";
export { default as ArrowLeftCircle } from "./icons/arrow-left-circle";
export { default as ArrowLeft } from "./icons/arrow-left";
export { default as ArrowRightCircle } from "./icons/arrow-right-circle";
export { default as ArrowRight } from "./icons/arrow-right";
export { default as ArrowUpCircle } from "./icons/arrow-up-circle";
export { default as ArrowUpLeft } from "./icons/arrow-up-left";
export { default as ArrowUpRight } from "./icons/arrow-up-right";
export { default as ArrowUp } from "./icons/arrow-up";
export { default as AtSign } from "./icons/at-sign";
export { default as Award } from "./icons/award";
export { default as BarChart2 } from "./icons/bar-chart-2";
export { default as BarChart } from "./icons/bar-chart";
export { default as BatteryCharging } from "./icons/battery-charging";
export { default as Battery } from "./icons/battery";
export { default as BellOff } from "./icons/bell-off";
export { default as Bell } from "./icons/bell";
export { default as Bluetooth } from "./icons/bluetooth";
export { default as Bold } from "./icons/bold";
export { default as BookOpen } from "./icons/book-open";
export { default as Book } from "./icons/book";
export { default as Bookmark } from "./icons/bookmark";
export { default as Box } from "./icons/box";
export { default as Briefcase } from "./icons/briefcase";
export { default as Calendar } from "./icons/calendar";
export { default as CameraOff } from "./icons/camera-off";
export { default as Camera } from "./icons/camera";
export { default as Cast } from "./icons/cast";
export { default as CheckCircle } from "./icons/check-circle";
export { default as CheckSquare } from "./icons/check-square";
export { default as Check } from "./icons/check";
export { default as ChevronDown } from "./icons/chevron-down";
export { default as ChevronLeft } from "./icons/chevron-left";
export { default as ChevronRight } from "./icons/chevron-right";
export { default as ChevronUp } from "./icons/chevron-up";
export { default as ChevronsDown } from "./icons/chevrons-down";
export { default as ChevronsLeft } from "./icons/chevrons-left";
export { default as ChevronsRight } from "./icons/chevrons-right";
export { default as ChevronsUp } from "./icons/chevrons-up";
export { default as Chrome } from "./icons/chrome";
export { default as Circle } from "./icons/circle";
export { default as Clipboard } from "./icons/clipboard";
export { default as Clock } from "./icons/clock";
export { default as CloudDrizzle } from "./icons/cloud-drizzle";
export { default as CloudLightning } from "./icons/cloud-lightning";
export { default as CloudOff } from "./icons/cloud-off";
export { default as CloudRain } from "./icons/cloud-rain";
export { default as CloudSnow } from "./icons/cloud-snow";
export { default as Cloud } from "./icons/cloud";
export { default as Code } from "./icons/code";
export { default as Codepen } from "./icons/codepen";
export { default as Codesandbox } from "./icons/codesandbox";
export { default as Coffee } from "./icons/coffee";
export { default as Columns } from "./icons/columns";
export { default as Command } from "./icons/command";
export { default as Compass } from "./icons/compass";
export { default as Copy } from "./icons/copy";
export { default as CornerDownLeft } from "./icons/corner-down-left";
export { default as CornerDownRight } from "./icons/corner-down-right";
export { default as CornerLeftDown } from "./icons/corner-left-down";
export { default as CornerLeftUp } from "./icons/corner-left-up";
export { default as CornerRightDown } from "./icons/corner-right-down";
export { default as CornerRightUp } from "./icons/corner-right-up";
export { default as CornerUpLeft } from "./icons/corner-up-left";
export { default as CornerUpRight } from "./icons/corner-up-right";
export { default as Cpu } from "./icons/cpu";
export { default as CreditCard } from "./icons/credit-card";
export { default as Crop } from "./icons/crop";
export { default as Crosshair } from "./icons/crosshair";
export { default as Database } from "./icons/database";
export { default as Delete } from "./icons/delete";
export { default as Disc } from "./icons/disc";
export { default as DivideCircle } from "./icons/divide-circle";
export { default as DivideSquare } from "./icons/divide-square";
export { default as Divide } from "./icons/divide";
export { default as DollarSign } from "./icons/dollar-sign";
export { default as DownloadCloud } from "./icons/download-cloud";
export { default as Download } from "./icons/download";
export { default as Dribbble } from "./icons/dribbble";
export { default as Droplet } from "./icons/droplet";
export { default as Edit2 } from "./icons/edit-2";
export { default as Edit3 } from "./icons/edit-3";
export { default as Edit } from "./icons/edit";
export { default as ExternalLink } from "./icons/external-link";
export { default as EyeOff } from "./icons/eye-off";
export { default as Eye } from "./icons/eye";
export { default as Facebook } from "./icons/facebook";
export { default as FastForward } from "./icons/fast-forward";
export { default as Feather } from "./icons/feather";
export { default as Figma } from "./icons/figma";
export { default as FileMinus } from "./icons/file-minus";
export { default as FilePlus } from "./icons/file-plus";
export { default as FileText } from "./icons/file-text";
export { default as File } from "./icons/file";
export { default as Film } from "./icons/film";
export { default as Filter } from "./icons/filter";
export { default as Flag } from "./icons/flag";
export { default as FolderMinus } from "./icons/folder-minus";
export { default as FolderPlus } from "./icons/folder-plus";
export { default as Folder } from "./icons/folder";
export { default as Framer } from "./icons/framer";
export { default as Frown } from "./icons/frown";
export { default as Gift } from "./icons/gift";
export { default as GitBranch } from "./icons/git-branch";
export { default as GitCommit } from "./icons/git-commit";
export { default as GitMerge } from "./icons/git-merge";
export { default as GitPullRequest } from "./icons/git-pull-request";
export { default as GitHub } from "./icons/github";
export { default as Gitlab } from "./icons/gitlab";
export { default as Globe } from "./icons/globe";
export { default as Grid } from "./icons/grid";
export { default as HardDrive } from "./icons/hard-drive";
export { default as Hash } from "./icons/hash";
export { default as Headphones } from "./icons/headphones";
export { default as Heart } from "./icons/heart";
export { default as HelpCircle } from "./icons/help-circle";
export { default as Hexagon } from "./icons/hexagon";
export { default as Home } from "./icons/home";
export { default as Image } from "./icons/image";
export { default as Inbox } from "./icons/inbox";
export { default as Info } from "./icons/info";
export { default as Instagram } from "./icons/instagram";
export { default as Italic } from "./icons/italic";
export { default as Key } from "./icons/key";
export { default as Layers } from "./icons/layers";
export { default as Layout } from "./icons/layout";
export { default as LifeBuoy } from "./icons/life-buoy";
export { default as Link2 } from "./icons/link-2";
export { default as Link } from "./icons/link";
export { default as Linkedin } from "./icons/linkedin";
export { default as List } from "./icons/list";
export { default as Loader } from "./icons/loader";
export { default as Lock } from "./icons/lock";
export { default as LogIn } from "./icons/log-in";
export { default as LogOut } from "./icons/log-out";
export { default as Mail } from "./icons/mail";
export { default as MapPin } from "./icons/map-pin";
export { default as Map } from "./icons/map";
export { default as Maximize2 } from "./icons/maximize-2";
export { default as Maximize } from "./icons/maximize";
export { default as Meh } from "./icons/meh";
export { default as Menu } from "./icons/menu";
export { default as MessageCircle } from "./icons/message-circle";
export { default as MessageSquare } from "./icons/message-square";
export { default as MicOff } from "./icons/mic-off";
export { default as Mic } from "./icons/mic";
export { default as Minimize2 } from "./icons/minimize-2";
export { default as Minimize } from "./icons/minimize";
export { default as MinusCircle } from "./icons/minus-circle";
export { default as MinusSquare } from "./icons/minus-square";
export { default as Minus } from "./icons/minus";
export { default as Monitor } from "./icons/monitor";
export { default as Moon } from "./icons/moon";
export { default as MoreHorizontal } from "./icons/more-horizontal";
export { default as MoreVertical } from "./icons/more-vertical";
export { default as MousePointer } from "./icons/mouse-pointer";
export { default as Move } from "./icons/move";
export { default as Music } from "./icons/music";
export { default as Navigation2 } from "./icons/navigation-2";
export { default as Navigation } from "./icons/navigation";
export { default as Octagon } from "./icons/octagon";
export { default as Package } from "./icons/package";
export { default as Paperclip } from "./icons/paperclip";
export { default as PauseCircle } from "./icons/pause-circle";
export { default as Pause } from "./icons/pause";
export { default as PenTool } from "./icons/pen-tool";
export { default as Percent } from "./icons/percent";
export { default as PhoneCall } from "./icons/phone-call";
export { default as PhoneForwarded } from "./icons/phone-forwarded";
export { default as PhoneIncoming } from "./icons/phone-incoming";
export { default as PhoneMissed } from "./icons/phone-missed";
export { default as PhoneOff } from "./icons/phone-off";
export { default as PhoneOutgoing } from "./icons/phone-outgoing";
export { default as Phone } from "./icons/phone";
export { default as PieChart } from "./icons/pie-chart";
export { default as PlayCircle } from "./icons/play-circle";
export { default as Play } from "./icons/play";
export { default as PlusCircle } from "./icons/plus-circle";
export { default as PlusSquare } from "./icons/plus-square";
export { default as Plus } from "./icons/plus";
export { default as Pocket } from "./icons/pocket";
export { default as Power } from "./icons/power";
export { default as Printer } from "./icons/printer";
export { default as Radio } from "./icons/radio";
export { default as RefreshCcw } from "./icons/refresh-ccw";
export { default as RefreshCw } from "./icons/refresh-cw";
export { default as Repeat } from "./icons/repeat";
export { default as Rewind } from "./icons/rewind";
export { default as RotateCcw } from "./icons/rotate-ccw";
export { default as RotateCw } from "./icons/rotate-cw";
export { default as Rss } from "./icons/rss";
export { default as Save } from "./icons/save";
export { default as Scissors } from "./icons/scissors";
export { default as Search } from "./icons/search";
export { default as Send } from "./icons/send";
export { default as Server } from "./icons/server";
export { default as Settings } from "./icons/settings";
export { default as Share2 } from "./icons/share-2";
export { default as Share } from "./icons/share";
export { default as ShieldOff } from "./icons/shield-off";
export { default as Shield } from "./icons/shield";
export { default as ShoppingBag } from "./icons/shopping-bag";
export { default as ShoppingCart } from "./icons/shopping-cart";
export { default as Shuffle } from "./icons/shuffle";
export { default as Sidebar } from "./icons/sidebar";
export { default as SkipBack } from "./icons/skip-back";
export { default as SkipForward } from "./icons/skip-forward";
export { default as Slack } from "./icons/slack";
export { default as Slash } from "./icons/slash";
export { default as Sliders } from "./icons/sliders";
export { default as Smartphone } from "./icons/smartphone";
export { default as Smile } from "./icons/smile";
export { default as SortAsc } from "./icons/sort-asc";
export { default as SortDesc } from "./icons/sort-desc";
export { default as Speaker } from "./icons/speaker";
export { default as Square } from "./icons/square";
export { default as Star } from "./icons/star";
export { default as StopCircle } from "./icons/stop-circle";
export { default as Sun } from "./icons/sun";
export { default as Sunrise } from "./icons/sunrise";
export { default as Sunset } from "./icons/sunset";
export { default as Tablet } from "./icons/tablet";
export { default as Tag } from "./icons/tag";
export { default as Target } from "./icons/target";
export { default as Terminal } from "./icons/terminal";
export { default as Thermometer } from "./icons/thermometer";
export { default as ThumbsDown } from "./icons/thumbs-down";
export { default as ThumbsUp } from "./icons/thumbs-up";
export { default as ToggleLeft } from "./icons/toggle-left";
export { default as ToggleRight } from "./icons/toggle-right";
export { default as Tool } from "./icons/tool";
export { default as Trash2 } from "./icons/trash-2";
export { default as Trash } from "./icons/trash";
export { default as Trello } from "./icons/trello";
export { default as TrendingDown } from "./icons/trending-down";
export { default as TrendingUp } from "./icons/trending-up";
export { default as Triangle } from "./icons/triangle";
export { default as Truck } from "./icons/truck";
export { default as Tv } from "./icons/tv";
export { default as Twitch } from "./icons/twitch";
export { default as Twitter } from "./icons/twitter";
export { default as Type } from "./icons/type";
export { default as Umbrella } from "./icons/umbrella";
export { default as Underline } from "./icons/underline";
export { default as Unlock } from "./icons/unlock";
export { default as UploadCloud } from "./icons/upload-cloud";
export { default as Upload } from "./icons/upload";
export { default as UserCheck } from "./icons/user-check";
export { default as UserMinus } from "./icons/user-minus";
export { default as UserPlus } from "./icons/user-plus";
export { default as UserX } from "./icons/user-x";
export { default as User } from "./icons/user";
export { default as Users } from "./icons/users";
export { default as VideoOff } from "./icons/video-off";
export { default as Video } from "./icons/video";
export { default as Voicemail } from "./icons/voicemail";
export { default as Volume1 } from "./icons/volume-1";
export { default as Volume2 } from "./icons/volume-2";
export { default as VolumeX } from "./icons/volume-x";
export { default as Volume } from "./icons/volume";
export { default as Watch } from "./icons/watch";
export { default as WifiOff } from "./icons/wifi-off";
export { default as Wifi } from "./icons/wifi";
export { default as Wind } from "./icons/wind";
export { default as XCircle } from "./icons/x-circle";
export { default as XOctagon } from "./icons/x-octagon";
export { default as XSquare } from "./icons/x-square";
export { default as X } from "./icons/x";
export { default as Youtube } from "./icons/youtube";
export { default as ZapOff } from "./icons/zap-off";
export { default as Zap } from "./icons/zap";
export { default as ZoomIn } from "./icons/zoom-in";
export { default as ZoomOut } from "./icons/zoom-out";
