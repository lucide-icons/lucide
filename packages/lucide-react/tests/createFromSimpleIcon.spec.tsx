import { describe, it, expect } from 'vitest';
import createFromSimpleIcon from '../src/createFromSimpleIcon';
import type { SimpleIcon } from 'simple-icons';
import { render } from '@testing-library/react';

describe('Using createFromSimpleIcon', () => {
  it('should create a component from a SimpleIcon', () => {
    const mockSimpleIcon: SimpleIcon = {
      title: 'Test Icon',
      slug: 'testicon',
      hex: '000000',
      source: 'https://test.com',
      svg: '<svg>...</svg>',
      path: 'M10 10 H 90 V 90 H 10 L 10 10',
    };

    const TestIcon = createFromSimpleIcon('TestIcon', mockSimpleIcon);

    const { container } = render(<TestIcon />);

    expect(container.firstChild).toMatchSnapshot();
    expect(container.firstChild).toBeDefined();
  });
});
