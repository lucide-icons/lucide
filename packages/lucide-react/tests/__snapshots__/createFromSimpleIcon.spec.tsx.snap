// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Using createFromSimpleIcon > should create a component from a SimpleIcon 1`] = `
<svg
  class="lucide lucide-test-icon"
  fill="none"
  height="24"
  stroke="currentColor"
  stroke-linecap="round"
  stroke-linejoin="round"
  stroke-width="2"
  viewBox="0 0 24 24"
  width="24"
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    d="M10 10 H 90 V 90 H 10 L 10 10"
    fill="currentColor"
  />
</svg>
`;
